### PR TITLE
chore: Add test for default value in binding object [TECH-1592]

### PR DESCRIPTION
**Pull request by enricocolasante**
Weirdly binding of default values in binding object done by Spring MVC seems to work fine.
I am just adding a Controller test to verify that.

### CODE REVIEW

File: dhis-2/dhis-web-api/src/test/java/org/hisp/dhis/webapi/controller/SpringBindingTest.java [hunk 1]
```diff
@@ -135,6 +135,14 @@ void shouldReturnABadRequestWhenInvalidValueForBooleanInBindingObjectIsPassedAsP
                         "Value 'INVALID' is not valid for parameter booleanValue. Invalid boolean value [INVALID]")));
   }
 
+  @Test
+  void shouldReturnADefaultValueWhenDefaultParameterIsNotPassed() throws Exception {
```
Comment (muilpp) [inline dhis-2/dhis-web-api/src/test/java/org/hisp/dhis/webapi/controller/SpringBindingTest.java:139]: could we also add a test to verify that the default value is not used if the parameter has a value?

File: dhis-2/dhis-web-api/src/test/java/org/hisp/dhis/webapi/controller/SpringBindingTest.java [hunk 2]
```diff
@@ -135,6 +135,22 @@ void shouldReturnABadRequestWhenInvalidValueForBooleanInBindingObjectIsPassedAsP
...
+    mockMvc
+        .perform(get(ENDPOINT + "/default").param("booleanValue", "true"))
```
Comment (teleivo) [inline dhis-2/dhis-web-api/src/test/java/org/hisp/dhis/webapi/controller/SpringBindingTest.java:141]: Why do you pass `booleanValue` `true`? You are not asserting on anything it has an influence, right? If that is the case I would remove it.

**Review by teleivo — COMMENTED** [review]
Could it be that there was a bug in Spring?

Or is it that the test setup is different from our production code (web config, ...) so that you are not seeing the same behavior?

Just wondering as you sounded puzzled as to why you cannot see it anymore 😬

**Comment by enricocolasante** [comment]
> Could it be that there was a bug in Spring?
> 
> Or is it that the test setup is different from our production code (web config, ...) so that you are not seeing the same behavior?
> 
> Just wondering as you sounded puzzled as to why you cannot see it anymore grimacing

I have no idea.
First thing I was trying to go back and check exactly that, if we changed Spring version or any related library, but going back in time and I couldn't reproduce locally the issue.

I was always testing it manually and if Marc wasn't there as well I would start to think that I got it all wrong and there was never an issue in the first place

**Pull request merged**